wicket-extensions/src/main/java/org/apache/wicket/extensions/ajax/markup/html/form/upload/UploadProgressBar.java

```diff
@@ -112,14 +112,38 @@ public void destroy(final Application application)
 
 	private static final long serialVersionUID = 1L;
 
-	private final Form<?> form;
+	private Form<?> form;
 
 	private MarkupContainer statusDiv;
 
 	private MarkupContainer barDiv;
 
 	private final FileUploadField uploadField;
 
+	/**
+	 * Constructor that will display the upload progress bar for every submit of the given form.
+	 *
+	 * @param id
+	 *            component id (not null)
+	 * @param uploadField
+	 *            the file upload field to check for a file upload, or null to display the upload
+	 *            field for every submit of the given form
+	 */
+	public UploadProgressBar(final String id,  final FileUploadField uploadField)
+	{
+		super(id);
```

Comment:
You can call `this(id, null, Args.nonNull(uploadField, "uploadField"))`
And move the non-null Form from https://github.com/apache/wicket/pull/571/files#diff-f75b5636ac8b2157552b9755243a10c168527c2132bd8ce61393386a49482047R183 to https://github.com/apache/wicket/pull/571/files#diff-f75b5636ac8b2157552b9755243a10c168527c2132bd8ce61393386a49482047R157